functions/zoom-uploader.py

```diff
@@ -288,19 +281,43 @@ def ingest(self):
             ("spatial", (None, "Zoom"))
         ]
 
-        for s3_filename in video_files:
-            url = self._generate_presigned_url(s3_filename)
-            params.extend([
-                ("flavor", (None, escape(ZOOM_OPENCAST_FLAVOR))),
-                ("mediaUri", (None, url))
-            ])
+        file_params = self._generate_file_params()
+        if not len(file_params):
+            raise Exception("No recording file view types that we want")
+        else:
+            params.extend(file_params)
 
```

Comment:
@lbjay, the "multipart/chunked+source" is a minimum upload requirement (i.e. presenter). Is it ever possible that there would be a "shared_screen" and "gallery_view" but no "active_speaker"? 
If so, is it possible to fallback one of the  "shared_screen" or "gallery_view" to be flavor "multipart/chunked+source" on the upload to protect from errors?

Reply:
_Is it ever possible that there would be a "shared_screen" and "gallery_view" but no "active_speaker"?_ -> Yes

Reply:
Yes, Opencast requires at least the "presenter" file so, if there's only shared_screen and gallery, one of them needs to be uploaded as the multipart/chunked+source.
Also: If there's no "presentation", the gallery needs to be sent as the presentation/chunked+source. This is because the workflow cannot test all the possible combinations so if there's 1 video, it needs to be presenter; if there are 2, they need to be presenter & presentaion.